### DISCUSSION TITLE
Does ANNIS support MySQL or Postgrers?

### DISCUSSION THREAD

**Discussion by fishfree** [korpling/ANNIS#826] [General]
I see only H2 supported, which is not an production-ready persistent solution.

**Comment by thomaskrause** [comment]
The H2 database is only used currently for storing the URL shortener entries, which is deactivated per default and I plan to deprecate this feature and replace it with a better mechanism in the future.
In the documentation (http://korpling.github.io/ANNIS/4.10/user-guide/configuration/index.html) there is an example how to enable the URL shortener and also how to adjust the configuration to support a persistent H2 database via the `spring.datasource.url` parameter.

Also, since we use Spring JPA for communicating with the database, replacing H2 with other databases supported by Spring JPA should be a simple configuration change, e.g. by updating `spring.datasource.url` and making sure that the correct dependency is included. Adding additional dependencies for MySQL or PostgreSQL drivers just before having an actual use case for this adds maintenance burden that seems quite large in comparison to its benefits.
